src/main/java/com/google/sps/data/Contest.java

```diff
@@ -26,41 +26,59 @@
 public abstract class Contest {
   public abstract String getName();
 
-  public abstract HashMap<Long, Candidate> getCandidates();
+  // This Contest references a collection of Candidate entities in Datastore. This HashSet
+  // represents their Key names.
+  public abstract HashSet<Long> getCandidates();
 
   public abstract String getDescription();
 
   public static Builder builder() {
     return new AutoValue_Contest.Builder();
   }
 
-  // creates a new Contest object by extracting the properties from "obj"
-  public static Contest fromJSONObject(JSONObject obj) throws JSONException {
-    return Contest.builder()
-        .setName(obj.getString("office"))
-        // TODO(caseyprice): get values for candidates and description
-        .setCandidates(new HashMap<Long, Candidate>())
-        .setDescription("")
-        .build();
-  }
-
-  // creates a new Entity and sets the proper properties.
-  public Entity toEntity() {
-    Entity entity = new Entity("Contest");
-    entity.setProperty("name", this.getName());
-    entity.setProperty("candidates", new ArrayList<String>());
-    entity.setProperty("description", this.getDescription());
-    return entity;
-  }
-
   @AutoValue.Builder
   public abstract static class Builder {
     public abstract Builder setName(String name);
 
-    public abstract Builder setCandidates(HashMap<Long, Candidate> candidates);
+    public abstract Builder setCandidates(HashSet<Long> candidates);
 
     public abstract Builder setDescription(String description);
 
     public abstract Contest build();
   }
+
+  // Creates a new Contest object by extracting the properties from "contestData". For each of its
+  // "candidate" properties, creates a new Candidate object and inserts it into the given Datastore
+  // instance.
+  public static Contest fromVoterInfoQuery(DatastoreService datastore, JSONObject contestData)
+      throws JSONException {
+    HashSet<Long> candidateKeyIds = new HashSet<>();
```

Comment:
Set<Long>? I think this can be treated as a more generic Set interface after the initialization.

Reply:
Same below.

Reply:
Also if we take Julian's suggestion, this would be ImmutableSet instead of Set, but it's the same idea (we shouldn't care later if this is a TreeSet or a HashSet underlying).